File: dotnet/Queries/AddTwoNumbers/AddTwoNumbersQueryHandler.cs
using System.Threading;
using System.Threading.Tasks;
using MediatR;

namespace dotnet.Queries.AddTwoNumbers {
    public class AddTwoNumbersQueryHandler : IRequestHandler<AddTwoNumbersQuery, int>
    {
        public async Task<int> Handle(AddTwoNumbersQuery request, CancellationToken cancellationToken)
        {
            return request.FirstNumber + request.SecondNumber;
        }
    }
}

File: java/foobar/src/main/java/foobar/queries/addtwonumbers/AddTwoNumbersQueryHandler.java
package foobar.queries.addtwonumbers;

import org.springframework.stereotype.Component;

import an.awesome.pipelinr.Command;

@Component
public class AddTwoNumbersQueryHandler implements Command.Handler<AddTwoNumbersQuery, Integer> {
    @Override
    public Integer handle(AddTwoNumbersQuery command) {
        return command.firstNumber + command.secondNumber;
    }
}